server/pulp/server/controllers/repository.py

```diff
@@ -1198,13 +1198,18 @@ def check_publish(repo_obj, dist_id, dist_inst, transfer_repo, conduit, call_con
         skip_for_predistributor = published_after_predistributor or not \
             predistributor_last_published
 
-    same_override = dist.last_override_config == config_override
-    if not same_override:
+    if not (dist.last_override_config == config_override):
         # Use raw pymongo not to fire the signal hander
         model.Distributor.objects(
             repo_id=repo_obj.repo_id,
             distributor_id=dist_id).update(set__last_override_config=config_override)
 
+    # do not compare force_full from config override - this particular difference itself does not
+    # determine the need of repo publish
+    config_override = {k: v for k, v in config_override.items() if k != 'force_full'}
+    dist.last_override_config = {k: v for k, v in dist.last_override_config.items()
+                                 if k != 'force_full'}
+    same_override = dist.last_override_config == config_override
```

Comment:
I would prefer if we used the same strategy we use in the other place we do this

https://github.com/pulp/pulp/blob/b8f5498f90869e14c0a72cfff5f04ae309d58885/server/pulp/server/controllers/repository.py#L947-L954

Reply:
Or reuse the same code

Reply:
@dralley , the other place is for an importer override, here it's the distributor's override config. Or is your suggestion to unify it somehow? I agree it can be improved, however in support of touch/change less in pulp 2, this PR looks ok to me. Let me know if I misunderstood or if you are in favor of improving anything here.

Reply:
I'm fine with not unifying it (to avoid changes) but I do think we should use the same strategy, since it's marginally cleaner, and would make it more obvious that the same thing is happening.